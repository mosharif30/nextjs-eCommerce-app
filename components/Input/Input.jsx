/* eslint-disable react/prop-types */
import React from 'react'

/** @jsxImportSource @emotion/react */
import { css } from '@emotion/react'
import { useTheme } from '@emotion/react'

function Input({ type = 'text', placeholder }) {
  const theme = useTheme()
  // console.log(theme.typography.h1)
  return (
    <input
      type={type}
      placeholder={placeholder}
      css={css`
        width: 100%;
        border: ${theme.border[1]} solid ${theme.borderColor.light};
        background-color: ${theme.inputColor.light};
        margin: ${theme.spaces[0]};
        outline: ${theme.spaces[0]};
        font-size: ${theme.typography.paragraph.fontSize};
        color: ${theme.colors.text};
        padding: ${theme.spaces[3]};
        &:focus {
          border: ${theme.border[1]} solid ${theme.colors.primary};
        }
      `}
    />
  )
}

export default Input
