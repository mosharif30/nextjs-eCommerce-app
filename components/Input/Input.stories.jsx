import React from 'react'

/** @jsxImportSource @emotion/react */
import { css } from '@emotion/react'
import { useTheme } from '@emotion/react'
import InputText from '../InputText/InputText'
import InputEmail from '../InputEmail/InputEmail'
import InputPassword from '../InputPassword/InputPassword'
import { Space } from '../Typography'
export default { title: 'Input' }
export const Input = () => (
  <div
    css={css`
      width: 30%;
      margin: 0 auto;
      direction: rtl;
    `}
  >
    <InputText placeholder="نام خود را وارد کنید" />
    <Space />
    <InputEmail placeholder="Email" />
    <Space />
    <InputPassword placeholder="پسورد" />
  </div>
)
