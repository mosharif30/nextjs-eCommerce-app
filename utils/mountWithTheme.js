import { ThemeProvider } from '@emotion/react'
import theme from '../configs/theme'
import renderer from 'react-test-renderer'

const mountWithTheme = (children) => {
  return renderer.create(
    <ThemeProvider theme={theme}>{children}</ThemeProvider>
  )
}
export default mountWithTheme
