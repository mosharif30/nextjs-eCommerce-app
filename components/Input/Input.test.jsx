import mountWithTheme from '../../utils/mountWithTheme'
import Input from './Input'
describe('Input', () => {
  test('render correctly', () => {
    const component = mountWithTheme(<Input>sample Button</Input>)
    const tree = component
    expect(tree).toMatchSnapshot()
  })
  test('render correctly placeholder', () => {
    const component = mountWithTheme(<Input placeholder={"sample placeholder"}>sample Button</Input>)
    const tree = component
    expect(tree).toMatchSnapshot()
  })
})
