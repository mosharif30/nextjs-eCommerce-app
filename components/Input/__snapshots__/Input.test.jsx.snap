// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input render correctly 1`] = `
Array [
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".css-i3lbul-Input{width:100%;word-break:break-word;border:2px solid #ecf0f1;background-color:#ecf0f1;margin:0px;outline:2px;font-size:1em;color:#2c3e50;padding:0px;height:32px;}.css-i3lbul-Input:focus{border-bottom:4px solid #1abc9c;background-color:#f5f6fa;}",
      }
    }
    data-emotion="css i3lbul-Input"
  />,
  <input
    aria-label="input"
    className="css-i3lbul-Input"
    onChange={[Function]}
    type="text"
  />,
]
`;

exports[`Input render correctly placeholder 1`] = `
Array [
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".css-i3lbul-Input{width:100%;word-break:break-word;border:2px solid #ecf0f1;background-color:#ecf0f1;margin:0px;outline:2px;font-size:1em;color:#2c3e50;padding:0px;height:32px;}.css-i3lbul-Input:focus{border-bottom:4px solid #1abc9c;background-color:#f5f6fa;}",
      }
    }
    data-emotion="css i3lbul-Input"
  />,
  <input
    aria-label="input"
    className="css-i3lbul-Input"
    onChange={[Function]}
    placeholder="sample placeholder"
    type="text"
  />,
]
`;
